wasm/store.go

```diff
@@ -757,6 +758,23 @@ func (s *Store) buildExportInstances(module *Module, target *ModuleInstance) (ro
 	return
 }
 
+// ValidateAddrRange checks if the given address range is a valid address range.
+// It accepts rangeSize as uint64 so that callers can add or multiply two uint32 addresses
+// without overflow. eg.) ValidateAddrRange(uint32Offset, uint64(uint32Size) + 1)
+func (m *MemoryInstance) ValidateAddrRange(addr uint32, rangeSize uint64) bool {
+	return uint64(addr) < uint64(len(m.Buffer)) && rangeSize <= uint64(len(m.Buffer))-uint64(addr)
```

Comment:
Can `rangeSize` be zero here (`from the example it looks like not`)? If not, we just need `uint64(len(m.Buffer))-uint64(addr) >= rangeSize` condition.

If rangeSize can be 0 you can do 

```
if diffSize := uint64(len(m.Buffer))-uint64(addr); diffSize > 0 && diffSize >= rangeSize {
   return true
}

return false
```

Reply:
This will result in underflow when `addr` is bigger than `len(m.Buffer)`.
`uint64(len(m.Buffer))-uint64(addr)`

Reply:
Yeah, if that happens the difference `uint64(len(m.Buffer))-uint64(addr)` is negative but that will never be `>=` than rangeSize which is a `uint64`.

Reply:
sorry, maybe you've lost me. 

>  uint64(len(m.Buffer))-uint64(addr) is negative 

IIUC, this never be negative because it's unsigned. For example, `uint64(1) - `uint64(2)` is `18446744073709551615` and will be  bigger than almost any `rangeSize`